traitlets/config/configurable.py

```diff
@@ -203,8 +210,13 @@ def _config_changed(self, change):
         section_names = self.section_names()
         self._load_config(change.new, traits=traits, section_names=section_names)
 
-    def update_config(self, config):
-        """Update config and load the new values"""
+    def update_config(self, config, skip_env=False):
```

Comment:
> That's a big effort. But why?

Because it breaks expectations of existing code, that `.update_config` will load the config given.

> For those wanting the new functionality, the current defaults makes more sense.

I disagree. `update_config(config)` loading the config given makes more sense to me.

>it should come preloaded with all hassle-free defaults, no?

Yes, though I think your assessment of hassle-free is the reverse of mine. Plus backward-compatibility is of absolute utmost priority for traitlets. Changing the default behavior of `.update_config` to have no effect if a lower priority config source is given is not better than preserving the existing behavior of respecting the arguments of the function call. User calls to `.update_config(config)` not loading that config is a backward-incompatible change and more surprising behavior.

The only calls to `.update_config` that ought to have lower priority than environment variables are those in `Application.load_config_file`. That should not be true for all config objects in general loaded at any time.

The default in this PR would require updates to IPython, because the only calls to `.update_config` in IPython/Jupyter projects are explicit user-provided high priority calls that should override environment variables. If the default were `override_env=True` and `Application.load_config_file` set `override_env=False`, then everything would behave correctly and nothing would need updating.